src/server/system_services/system_server.js

```diff
@@ -1550,6 +1550,97 @@ function get_entity_info(entity, entity_type) {
             sys.name === entity.unwrap())) || undefined;
 }
 
+async function upgrade_master_keys() {
+    let master_keys = [];
+    let buckets_updates = [];
+    let accounts_updates = [];
+    let pools_updates = [];
+    let namespace_resources_updates = [];
+    let system_master_key = system_store.data.systems[0].master_key_id;
+    // upgrade system master key if it doesn't exist
+    if (!system_master_key) {
+        system_master_key = system_store.master_key_manager.new_master_key({
+            description: `master key of ${system_store.data.systems[0]._id.toString()} system`,
+            master_key_id: system_store.master_key_manager.get_root_key_id(),
+            cipher_type: system_store.master_key_manager.get_root_key().cipher_type
+        });
+        master_keys.push(system_master_key);
+    }
+    // upgrade buckets master keys
+    _.map(system_store.data.buckets, bucket => {
+        if (bucket.master_key_id) return;
+        const bucket_m_key = system_store.master_key_manager.new_master_key({
+            description: `master key of ${bucket._id} bucket`,
+            master_key_id: system_master_key._id,
+            cipher_type: system_master_key.cipher_type
+        });
+        buckets_updates.push({
```

Comment:
Do we want to encrypt all of the buckets and their chunks upon upgrade?
I remember that we've discussed not re-encrypting data (in that case it is the chunk's cipher), but the question still remains.

Reply:
yes, after discussion we decided to keep this flow.